src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/Type.java

```diff
@@ -415,15 +413,6 @@
      * @return a new pointer type with given pointee type.
      */
     static Type.Delegated pointer(Type pointee) {
-        return new TypeImpl.PointerImpl(() -> pointee);
-    }
-
-    /**
-     * Creates a new pointer type with given pointee type.
-     * @param pointee factory to (lazily) build the pointee type.
-     * @return a new pointer type with given pointee type (lazily built from factory).
-     */
-    static Type.Delegated pointer(Supplier<Type> pointee) {
```

Comment:
This shouldn't be removed! It's the only way for a client to create mutually referring pointer types.

Reply:
This change is what I meant at beginning to make this an implementation detail.

I have trouble to figure out how to use this API without an API providing something like the TypeReference, which I can implement as a Type that is a transparent delegate.

Imagine Type.declare(Declareation.struct(pos, "foo", Type.pointer(<what can I do to provide this struct type>)).

Reply:
The problem is always with cases like:

```
struct A {
  struct B *b;
};

struct B {
  struct A *a;
};
```

If you have a supplier-accepting constructor, you can imagine an approach where you store the pointers in different fields, and then you set up the lambdas to retrieve the field values - that will work in such pathological cases.